### PR TITLE
compat unsafe_string to deprecate bytestring, compat os_only macros

### PR DESCRIPTION
**Pull request by wookay**
hello, Fixed deprecated bytestring and os related macros.

### CODE REVIEW

**Comment by tkelman** [comment]
thank you! lgtm if it passes CI

File: src/ZMQ.jl [hunk 1]
```diff
@@ -4,7 +4,7 @@ VERSION >= v"0.4.0-dev+6521" && __precompile__(true)
 
 module ZMQ
 using Compat
-import Compat.String
+import Compat: String, unsafe_string
```
Comment (yuyichao) [inline src/ZMQ.jl]: This shouldn't be necessary.

File: src/ZMQ.jl [hunk 2]
```diff
@@ -43,7 +43,7 @@ export
     #Types
     StateError,Context,Socket,Message,
     #functions
-    set, subscribe, unsubscribe,
+    set, subscribe, unsubscribe, unsafe_string,
```
Comment (yuyichao) [inline src/ZMQ.jl]: Neither does this.

**Comment by yuyichao** [comment]
LGTM too other than the import/export

File: src/ZMQ.jl (hunk 1 above)
Comment (wookay) [inline src/ZMQ.jl]: when `import Compat: unsafe_string`, I got
```
julia> using ZMQ
INFO: Recompiling stale cache file /Users/wookyoung/.julia/lib/v0.5/ZMQ.ji for module ZMQ.

julia> method_exists(unsafe_string, (Message,))
true
```

but without `import Compat: unsafe_string`, got
```
julia> using ZMQ
INFO: Recompiling stale cache file /Users/wookyoung/.julia/lib/v0.5/ZMQ.ji for module ZMQ.

julia> method_exists(unsafe_string, (Message,))
false
```

and test has also failed,
```
Testing with ZMQ version 3.2.5
ERROR: LoadError: AssertionError: typeof(ex) == StateError
 in include_from_node1(::String) at ./loading.jl:426
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /Users/wookyoung/.julia/v0.5/ZMQ/test/runtests.jl, in expression starting on line 16
```

Comment (yuyichao) [inline src/ZMQ.jl]: Ah, I didn't realize that ZMQ is extending `unsafe_string`. In that case, the export should still be removed and I suspect the `@deprecate` should be conditional on 0.5 to not get method override and depwarn on 0.4-

Comment (yuyichao) [inline src/ZMQ.jl]: NVM, the `@deprecate` seems fine since the `Compat.unsafe_string` is not an alias. So only the export one.

Comment (wookay) [inline src/ZMQ.jl]: ok. :)

**Comment by wookay** [comment]
it needs to rebase before to be merged. thanks.

**Pull request merged**